tom-swifty/SwiftyOptions.cs

```diff
@@ -261,38 +264,62 @@ public void CheckForOptionErrors (ErrorHandling errors)
 							var targetOS = targets [0].ClangTargetOS ();
 							if (SwiftGluePath != null) {
 								string path = null;
-								if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "mac/XamGlue.framework");
-								} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "iphone/XamGlue.framework");
-								} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "appletv/XamGlue.framework");
-								} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
-									path = Path.Combine (SwiftGluePath, "watch/XamGlue.framework");
+								if (!isLibrary) {
+									if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "mac/XamGlue.framework");
+									} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "iphone/XamGlue.framework");
+									} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "appletv/XamGlue.framework");
+									} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "watch/XamGlue.framework");
+									}
```

Comment:
AFAIK you can use a switch with pattern matching. https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching